runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -546,7 +546,7 @@ public String toString() {
      * @return an appropriate context
      */
     private ICorfuExecutionContext.Context getContext(ISMRStream stream) {
-        if (stream == optimisticStream) {
+        if (stream.equals(optimisticStream)) {
```

Comment:
This conversion is not always correct. One is a reference equality check and the other is a logical equality check. For example, this logic would return different results if there are two different objects (i.e, different references), but they are logically equal.

Reply:
Yeah I agree, but here `equals` could have the same  impact as `==` since we didn't override the `equals` method in either of `ISMRStream` implementation. It should be fine to keep this conversion here.